src/Lecture1.hs

```diff
@@ -104,7 +126,13 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: [Char] -> Int
+strSum str = sumList 0 (map (\x -> read (x) :: Int) (words str))
+    where 
+        sumList result list = 
+            if null list
+                then result
+                else sumList (result + head list) (tail list)
```

Comment:
Your implementation is absolutely correct 💯 💎 
And I see that you understand recursion perfectly!

To make your solution even more elegant, you can use the standard function `sum` that finds sum of all elements in the list.
Also, you don't need to specify the type of `read`. You can just use `map read`. You already specified in the type signature that `strSum` returns `Int` and due to type inference Haskell can use the correct type of `read` without explicit type annotation near `read`.

Reply:
Wow, thank you very much! 
And, first of all, thank you for all those elaborate and neat comments with a lot of useful information! I am very grateful for that!

Also, this is sooo much simpler and more beautiful after your corrections.